src/Illuminate/Http/Client/PendingRequest.php

```diff
@@ -958,11 +959,20 @@ protected function sendRequest(string $method, string $url, array $options = [])
 
         $laravelData = $this->parseRequestData($method, $url, $options);
 
+        $onStats = function ($transferStats) {
+            $this->transferStats = $transferStats;
+        };
+
+        if (data_get($this->options, 'on_stats') instanceof Closure) {
+            $onStats = function ($transferStats) use ($onStats) {
+                $onStats($transferStats);
+                call_user_func($this->options['on_stats'], $transferStats);
+            };
+        }
+
```

Comment:
Something like this is probably more logical (e.g. your callback can modify the transferStats. It will be assigned to `$this->transferStats` after completion):
```php
$onStats = function ($transferStats) {
    if (($callback = $this->options['on_stats'] ?? false) instanceof Closure) {
        $transferStats = $callback($transferStats);
    }
    $this->transferStats = $transferStats;
};
```

Reply:
Isn't this force the callback to return the transferStats? In my implementation we always have the transferStats assigned to the Response Object, so it wasn't changeable, but in the recommendation if I do not return the transferStats in my custom callback I don't have it right?

I changed it a little bit:
```php
$onStats = function ($transferStats) {
    if (($callback = $this->options['on_stats'] ?? false) instanceof Closure) {
        $transferStats = $callback($transferStats) ?? $transferStats;
    }
    $this->transferStats = $transferStats;
};
```
This should be better if we don't want to force a return on the custom callback